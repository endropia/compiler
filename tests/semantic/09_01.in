var
	a: array[0..10] of integer;
	b: array[0..10, 0..10] of integer;
	c: array[0..10] of array[0..10] of integer;

begin
	write(a[1], b[1][1], c[1][1]);
end.