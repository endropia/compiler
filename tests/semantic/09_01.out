program : Unnamed program
   var: 
      a
      array
      type: integer
      range
         0
         10
   var: 
      b
      array
      type: integer
      range
         0
         10
      range
         0
         10
   var: 
      c
      array
      array
         type: integer
         range
            0
            10
      range
         0
         10
   stmts:
      call
         write
            array
               a
               1
            array
               array
                  b
                  1
               1
            array
               array
                  c
                  1
               1

scope     name                          class               
------------------------------------------------------------
0         integer                       primitive type      
0         double                        primitive type      
0         boolean                       primitive type      
0         char                          primitive type      
0         string                        primitive type      
1         a                             variable            
1         b                             variable            
1         c                             variable            
